packages/flutter/lib/src/widgets/layout_builder.dart

```diff
@@ -168,10 +178,24 @@ mixin RenderConstrainedLayoutBuilder<ConstraintType extends Constraints, ChildTy
     markNeedsLayout();
   }
 
+  // Whether `_callback` must be called upon next layout.
+  //
+  // Normally the callback is only called when layout constraints change.
+  // However, we must also call the callback when widget updates, e.g. after
+  // `setState`, or `didChangeDependencies`.
+  bool _mustCallCallback = true;
+
+  // The constraints that were passed to the [ConstrainedLayoutBuilder] last time.
+  Constraints _previousConstraints;
+
   /// Invoke the layout callback.
   void layoutAndBuildChild() {
```

Comment:
This should be renamed "rebuildIfNecessary" or some such, since it does not in fact ever do layout, and also might not call build and may never create a child.

Reply:
Done.